resources/js/alpine/navbarSearchComponent/navbarSearchComponent.ts

```diff
@@ -0,0 +1,149 @@
+import { fetcher } from '@/utils';
+import { computePosition, autoUpdate, ReferenceElement } from '@floating-ui/dom';
+
+type SearchResult = {
+  label: string,
+  mylink: string,
+}
+
+type NavbarSearchComponentType = {
+  showSearchResults:boolean,
+  searchText:string,
+  results: SearchResult[],
+  selectedIndex:number,
+  activeDecendentId: string,
+  init: (formEl: ReferenceElement, ulEl:HTMLElement)=>void,
+  handleClickSearchResult: (label: string) => void,
+  handleKeyDown: (e:KeyboardEvent) => void,
+  handleKeyUp: (e:KeyboardEvent, el:HTMLInputElement) => void,
```

Comment:
There is a subtle type mismatch here. `handleKeyUp` is async, thus it will always return a Promise under the hood:

```suggestion
  handleKeyUp: (e:KeyboardEvent, el:HTMLInputElement) => Promise<void>,
```